### PR TITLE
Remove `generateUUIDv7(NonMonotonic|ThreadMonotonic)` functions

### PR DESCRIPTION
**Pull request by rschu1ze**
See https://github.com/ClickHouse/ClickHouse/pull/63577#issuecomment-2134941323

The two released functions are not in a released version yet, removing them is okay.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 52d77052d3ed3dd5c7ff56d246ba887717490c4b with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64506/52d77052d3ed3dd5c7ff56d246ba887717490c4b/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by pet74alex** [comment]
@rschu1ze i think the code needs to be changed to acquire the full block size range for the counter similar to `generateSnowFlakeID` (unfortunately the same technique with an atomic state is not possible here as timestamp and counter together occupy more than 64 bits).

**Comment by rschu1ze** [comment]
@pet74alex Agree, I opened #64507 for that

**Comment by rschu1ze** [comment]
@devcrafter Test failures are unrelated, approve? Thanks.

File: src/Functions/generateUUIDv7.cpp [hunk 1]
```diff
@@ -262,27 +225,16 @@ class FunctionGenerateUUIDv7Base : public TargetSpecific::Default::FunctionGener
...
 {
-    static constexpr auto doc_syntax_format = "{}([expression])";
-    static constexpr auto example_format = "SELECT {}()";
-    static constexpr auto multiple_example_format = "SELECT {f}(1), {f}(2)";
-
-    FunctionDocumentation::Description description = FillPolicy::description;
-    FunctionDocumentation::Syntax syntax = fmt::format(doc_syntax_format, FillPolicy::name);
+    FunctionDocumentation::Description description = R"(Generates a UUID of version 7. The generated UUID contains the current Unix timestamp in milliseconds (48 bits), followed by version "7" (4 bits), a counter (42 bit, including a variant field "2", 2 bit) to distinguish UUIDs within a millisecond, and a random field (32 bits). For any given timestamp (unix_ts_ms), the counter starts at a random value and is incremented by 1 for each new UUID until the timestamp changes. In case the counter overflows, the timestamp field is incremented by 1 and the counter is reset to a random new start value. Function generateUUIDv7 guarantees that the counter field within a timestamp increments monotonically across all function invocations in concurrently running threads and queries.)";
```
Comment (devcrafter) [inline src/Functions/generateUUIDv7.cpp:230]: Minor. Can we have multiline description here? It's 700+ length description to read

**Review by devcrafter — APPROVED** [review]

File: src/Functions/generateUUIDv7.cpp (hunk 1 above)
Comment (rschu1ze) [reply src/Functions/generateUUIDv7.cpp:230]: Sure ... since that applies to a related function too (`generateSnowflakeId`), I'd do this in a separate PR.

**Comment by rschu1ze** [comment]
ClickHouse AST Fuzzer (ubsan):
- `SELECT * FROM old_school_table__fuzz_30 PREWHERE toNullable(toLowCardinality(1)) ORDER BY generateUUIDv7(generateUUIDv7(materialize(toNullable(1)))) DESC NULLS LAST` causes `Logical error: 'RangeReader read 1 rows, but 8192 expected.'`.
- A simplified scenario with the same error is https://fiddle.clickhouse.com/bcf6a063-258b-4347-af89-c4d2f257f84e which is a dup of https://github.com/ClickHouse/ClickHouse/issues/56640

ClickHouse Integration Tests (aarch64) [5/6]: https://github.com/ClickHouse/ClickHouse/issues/64451

ClickHouse Upgrade Check (debug): not all settings are reflected in settings history, unrelated

ClickHouse Stateless Tests (coverage) [5/6]: https://github.com/ClickHouse/ClickHouse/issues/64593

A sync: no conflicts.

Lgtm, forcing MergeableCheck green.

**Pull request merged**